Change Height on Footer element in Bonfire View

FFX01 opened the pull request:
Refer to line 677. Proposing removal of Padding increase on footer links in bonfire view as it causes the page content to exceed viewport height and creates a scrollbar. This issue currently effects all bonfire views. In response to issue #1247 

closes #1247

FFX01 committed:
Remove padding on footer link hover 

Refer to line 677. Proposing removal of Padding increase on footer links in bonfire view as it causes the page content to exceed viewport height and creates a scrollbar. This issue currently effects all bonfire views.

LumenTeun commented:
I'm sorry, but I'm not a fan of this solutions since it changes the intended design of the footer when hovered over. The enitre height of the footer is supposed to change background, not just the icon/text background.

FFX01 commented:
Alright, let me see what I can do about it. Honestly, CSS is my weak point.

FFX01 committed:
second attempt at fixing view height issue

Reinstated padding on footer link hover to maintain same aesthetic behavior as before. Changed height property of footer element to max-height element to allow ful expansion without exceeding viewport height.

LumenTeun commented:
Please consier [rebasing](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) these commits into a single one.

FFX01 commented:
@LumenTeun Unfortunately, my work machine does not have git installed and I do not have admin access to install it. I can't actually rebase without the git command line as far as i know. would it be a bad idea to close this request and create a new one?

LumenTeun commented:
@FFX01 I guess you can just wait and rebase when you get to another computer. I'm not an owner of this repo, so it's not really my business, but opening and closing pull request lightly really bugs me ;)

FFX01 commented:
@LumenTeun That's what I thought. It would bother me as well. I'll do it tonight when I get home. Thanks for your input.

BerkeleyTrue commented:
@FFX01 Sorry for the delay.

I'm not completely sure what this is trying to fix. I've gone through the issue that it references but I cannot duplicate the problem. Do you mind adding some screen caps here of the problem and solution?

ahstro commented:
@BerkeleyTrue When scrolling to the very bottom of any FCC page, it usually looks like this:
![fcc-regular](https://cloud.githubusercontent.com/assets/3090888/9083517/2b5745c8-3b6b-11e5-8c15-4842ffa42df5.png)
But when you're hovering over one of the footer items, an extra pixel is added somehow, and allows you to scroll further down, like this:
![fcc-hover](https://cloud.githubusercontent.com/assets/3090888/9083518/2b5c5f18-3b6b-11e5-9168-6bb7be9a41ef.png)
On pages that are just under the height of someone's screen, this would have the added side effect of producing a scrollbar on the right, but it's an issue regardless.

AlexandroPerez commented:
I was looking at the styles using Developer tools, and it seems that the hover state of the  links get a `padding-bottom` of 14px, wich is 2px too much (adding that gap). I experimented with it, and changing `padding-bottom: 12px` removes the extra gap, while still getting a nice hover state. The `padding-top` has to remain at 14px though:

```
.fcc-footer a:hover {
    padding-top: 14px
    padding-bottom: 12px;
    ...
}
```

AlexandroPerez commented:
the following is what I think is the issue. Take a close look at the padding:

public/css/main.css

```
.fcc-footer {
  ...
  padding: 12px;
  ...
  a {
    ...
    &:hover {
      padding-top: 14px;
      padding-bottom: 14px;
      ...
    }
  }
}
```

I was trying to work on this issue locally, but I need help. I'd like it if someone could guide me through the git process. I forked my own copy, and I have it locally, so I'll work on the above style until it works, but I have no idea how to go beyond that point (syncing properly with frecodecamp's repository, what to do after I create the branch, and how to push properly)

Lin-H commented:
@AlexandroPerez Once you finish your branch on your copy, you can click the green button to submit a pull request
![20150808113546](https://cloud.githubusercontent.com/assets/7029374/9148886/4c8d0a0e-3dc1-11e5-9b6c-0fb36d872d94.png)
compare Free Code Camp's `staging` branch with your `user-xxx` branch. (`user-xxx` is the name of the branch you're working on)
Then create.

AlexandroPerez commented:
@Lin-H Ok, thanks. I'll give it a try once when I get the chance, though I'm wondering if I really should, since this has a tag of `blocked`

BerkeleyTrue commented:
@alexandroperez have at it

AlexandroPerez commented:
@BerkelyTrue I'll give it a try, thanks.

AlexandroPerez commented:
@BerkeleyTrue I just made a pull request. I hope I was able to fix the problem. I tested it on chrome and firefox using their tools.

Changes made:

`.fcc-footer {`
* Removed `height`: Height should be handled by `<a>` elements, so that their hover state can cover the entire height of the footer.
* `padding` set to 0: This allows `<a>` elements to take the entire height of the footer.

`.fcc-footer a {`
* `display` set to `inline-block` to allow `<a>` elements to have a `height` of 50px, making the `.fcc-footer` remain the same height as before.
* `padding` set to 12px top/bottom and 10px left/right. the padding that used to be in the `.fcc-footer` (12px top/bottom) is now added to the links instead (which use to have 0 top/bottom, and only 10px left/right).
* `height` of 50px: This handles the lack of height in the `.fcc-footer`, and links no longer have to rely on strange extra padding in their hover state to cover the entire footer.

`.fcc-footer a:hover {`
* Removed `padding-top` and `padding-bottom`. Is no longer necessary to rely on this weird trick to have the hover state cover the entire footer's height.

benmcmahon100 commented:
@BerkeleyTrue @QuincyLarson Yay or Nae on this do we think?

benmcmahon100 commented:
We fixed this in another pull request. Thanks for taking a look at this though folks!
